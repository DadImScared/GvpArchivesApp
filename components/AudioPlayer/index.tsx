import * as React from "react";
import { StyleSheet, View } from "react-native";

import { withAudioPlayer } from "../withAudioPlayer";
import ButtonGroup from "./ButtonGroup";
import DurationSeeker from "./DurationSeeker";
import { IAudioPlayerState } from "./Provider";
import Title from "./Title";

interface IProps {
  audio: IAudioPlayerState;
}

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: "white",
    bottom: 0,
    padding: 8,
    position: "absolute",
    width: "100%",
  }
});

export class AudioPlayer extends React.Component<IProps> {
  render() {
    const {
      audio: {
        isSeeking,
        isPlaying,
        onPlayPausePressed,
        getSongName,
        getSeekSliderPosition,
        getTimeStamp,
        onSeekSliderChange,
        onSeekSliderComplete

      }
    } = this.props;
    return (
      <View style={styles.wrapper}>
        <Title getSongName={getSongName} />
        <DurationSeeker
          isSeeking={isSeeking}
          getSeekSliderPosition={getSeekSliderPosition}
          getTimeStamp={getTimeStamp}
          onSeekSliderChange={onSeekSliderChange}
          onSeekSliderComplete={onSeekSliderComplete}
        />
        <ButtonGroup
          onPlayPausePressed={onPlayPausePressed}
          isPlaying={isPlaying}
        />
      </View>
    );
  }
}

export default withAudioPlayer(AudioPlayer) as any; // property error without "any"
