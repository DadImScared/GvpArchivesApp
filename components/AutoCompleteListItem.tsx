import * as React from "react";
import { Text } from "react-native";

import { ListItem } from "native-base";

interface IProps {
  text: string;
}

export class AutoCompleteListItem extends React.Component<IProps> {

  render() {
    return (
      <ListItem button={true} noIndent={true}>
        <Text>{this.props.text}</Text>
      </ListItem>
    );
  }
}

export default AutoCompleteListItem;
