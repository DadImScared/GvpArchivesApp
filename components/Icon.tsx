import * as React from "react";

import { Icon as NativeBaseIcon, NativeBase } from "native-base";

export class Icon extends React.Component<NativeBase.Icon> {
  render() {
    return (
      <NativeBaseIcon {...this.props} />
    );
  }
}

export default Icon;
