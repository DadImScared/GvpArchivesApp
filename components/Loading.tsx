import * as React from "react";

import { Spinner } from "native-base";
import { View } from "react-native";

export const Loading: React.FunctionComponent<{ isLoading: boolean }> = ({ isLoading }) => isLoading ? (
  <View>
    <Spinner />
  </View>
) : null;

export default Loading;
