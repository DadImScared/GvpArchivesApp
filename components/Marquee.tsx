import * as React from "react";

import Ticker, { ITextMarqueeProps } from "react-native-text-ticker";

const ForwardedMarquee = React.forwardRef<Ticker, ITextMarqueeProps>(({ children, ...other }, ref) => (
  <Ticker {...other} ref={ref}>
    {children}
  </Ticker>
));

export default ForwardedMarquee;
