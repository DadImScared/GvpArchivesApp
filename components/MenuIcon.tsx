import * as React from "react";
import { NavigationScreenProps } from "react-navigation";

import { HeaderItem, IoniconsHeaderButtons } from "../screens/Header";

export class MenuIcon extends React.Component<NavigationScreenProps> {
  onPress = () => {
    this.props.navigation.openDrawer();
  };

  render() {
    return (
      <IoniconsHeaderButtons>
        <HeaderItem iconName={"md-menu"} onPress={this.onPress} title={"Menu"} />
      </IoniconsHeaderButtons>
    );
  }
}

export default MenuIcon;
