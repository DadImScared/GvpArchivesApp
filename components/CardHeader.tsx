import * as React from "react";

import { Body, CardItem, Text, View } from "native-base";
import Icon from "./Icon";
import { routes } from "./SideBar";

interface IProps {
  item: any;
  extraSubHeader?: string;
}

export const CardHeader: React.FunctionComponent<IProps> = ({ item, extraSubHeader }) => (
  <CardItem header={true}>
    <Body>
      <View style={{ flexDirection: "row", justifyContent: "space-between", width: "100%", alignItems: "center" }}>
        <Text style={{ width: "80%" }}>{item.title.trim()}</Text>
        <Icon {...routes[item.category].icon} />
      </View>
      <Text note={true}>{routes[item.category].displayName}{extraSubHeader ? `${extraSubHeader}` : null}</Text>
    </Body>
  </CardItem>
);

export default CardHeader;
