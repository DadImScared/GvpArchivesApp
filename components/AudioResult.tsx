import * as React from "react";

import { Button, Card, CardItem } from "native-base";
import { StyleSheet, Text } from "react-native";

import { Item } from "../reducers/itemsById";
import CardHeader from "./CardHeader";
import Icon from "./Icon";

export interface IAudioResultProps {
  item: Item;
  [key: string]: any;
}

const styles: any = StyleSheet.create({
  button: {
    alignItems: "center",
    justifyContent: "space-between",
    margin: 4,
    minWidth: "30%"
  },
  buttonText: {
    color: "white",
    marginLeft: 16,
    marginRight: 16
  }
});

export const AudioResult: React.FC<IAudioResultProps> = ({ item }) => (
  <Card>
    <CardHeader item={item} />
    <CardItem style={{ justifyContent: "space-between" }} footer={true}>
      <Button primary={true} style={styles.button}>
        <Text style={styles.buttonText}>
          Play
        </Text>
        <Icon name={"play"} />
      </Button>
      <Button info={true} style={styles.button}>
        <Text style={styles.buttonText}> Add To Playlist</Text>
        <Icon type={"MaterialCommunityIcons"} name={"playlist-plus"} />
      </Button>
    </CardItem>
  </Card>
);

export default AudioResult;
