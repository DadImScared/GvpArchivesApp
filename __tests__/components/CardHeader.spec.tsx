import * as React from "react";
import * as renderer from "react-test-renderer";

import { CardHeader } from "../../components/CardHeader";

describe("CardHeader", () => {
  it("should render", () => {
    const tree = renderer.create(<CardHeader item={{ title: "Card title here", category: "song" }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with extraSubHeader", () => {
    const item = { title: "Card title here", category: "song" };
    const tree = renderer.create(<CardHeader item={item} extraSubHeader={"Continued text"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
