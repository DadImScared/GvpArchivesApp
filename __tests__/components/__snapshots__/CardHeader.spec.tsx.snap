// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader should render 1`] = `
<View
  header={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderBottomWidth": null,
      "borderRadius": 2,
      "flexDirection": "row",
      "paddingBottom": 15,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "width": "80%",
            },
          ]
        }
        uppercase={false}
      >
        Card title here
      </Text>
      <Icon
        name="music-note"
        type="MaterialCommunityIcons"
      />
    </View>
    <Text
      note={true}
      style={
        Object {
          "color": "#808080",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "200",
          "marginRight": 20,
        }
      }
      uppercase={false}
    >
      Songs
    </Text>
  </View>
</View>
`;

exports[`CardHeader should render with extraSubHeader 1`] = `
<View
  header={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderBottomWidth": null,
      "borderRadius": 2,
      "flexDirection": "row",
      "paddingBottom": 15,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "width": "80%",
            },
          ]
        }
        uppercase={false}
      >
        Card title here
      </Text>
      <Icon
        name="music-note"
        type="MaterialCommunityIcons"
      />
    </View>
    <Text
      note={true}
      style={
        Object {
          "color": "#808080",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "200",
          "marginRight": 20,
        }
      }
      uppercase={false}
    >
      Songs
      Continued text
    </Text>
  </View>
</View>
`;
